cmd/clusterctl/client/config.go

```diff
@@ -305,13 +305,19 @@ func (c *clusterctlClient) templateOptionsToVariables(options GetClusterTemplate
 	if options.ControlPlaneMachineCount < 1 {
 		return errors.Errorf("invalid ControlPlaneMachineCount. Please use a number greater or equal than 1")
 	}
-	c.configClient.Variables().Set("CONTROL_PLANE_MACHINE_COUNT", strconv.Itoa(options.ControlPlaneMachineCount))
+	// set CONTROL_PLANE_MACHINE_COUNT if it isn't set already
+	if _, err := c.configClient.Variables().Get("CONTROL_PLANE_MACHINE_COUNT"); err != nil {
```

Comment:
shouldn't this be checking `options.ControlPlaneMachineCount` like above for KubernetesVersion?

Reply:
do you mean checking that it isn't empty like we do for KubernetesVersion? The `options.ControlPlaneMachineCount` defaults to 1 so it will always be set.